lib/temp-directory.js

```diff
@@ -8,15 +8,16 @@ var uuid = require('uuid');
 var osenv = require('osenv');
 
 function create(context, next) {
-  var _path = path.join(osenv.tmpdir(),uuid.v4());
-  var npmConfigTmp = path.join(_path, 'npm_config_tmp');
-
-  context.path = _path;
-  context.npmConfigTmp = npmConfigTmp;
+  if (context.options && context.options.tmpDir) {
+    context.path = path.join(context.options.tmpDir, uuid.v4(), context.module.name);
+  } else {
+    context.path = path.join(osenv.tmpdir(),uuid.v4());
```

Comment:
missing space between tmpdir and uuid 

want to put together a pr with a lint rule to enforce this after this lands?